test-suite-data.json

```diff
@@ -347,6 +347,42 @@
     "subpath": null,
     "is_invalid": true
   },
+  {
+    "description": "simple vcpkg purl",
+    "purl": "pkg:vcpkg/ms-gsl@4.0.0?registry_version=4e2b371",
```

Comment:
Multiple version strings look confusing. If registry_version is more precise, then that could be used as the version.

Reply:
`registry_version` is more precise, but also less useful. The only meaningful operation on `registry_version` is equality -- whereas the "main" version can be ordered and partially match.

Reply:
Scripts could be developed to check if a given commit hash is part of a release tag.

Reply:
It's not a release tag, right? The registry version qualifier is something like a commit ID in the microsoft/vcpkg Git repo, but the package version is the value of the "version" property in the ports/ms-gsl/vcpkg.json file.

Even if you wanted to order by registry version and there existed an algorithm that could turn the commit ref into a release tag for both sides of a comparison (assuming there is one and it's orderable), such a script wouldn't work offline because it requires access to the repository, and it wouldn't be possible for users looking at the registry version to tell which version was newer or even if the versions were different at all.

There's precedence for doing it this way. `pkg:conan/ms-gsl@4.0.0?rrev=4e3b371` describes a packaging of ms-gsl version 4.0.0 built using a particular recipe. However, instead of a Git commit ID [it's a hash of the recipe files](https://docs.conan.io/en/1.46/cheatsheet.html#revisions). That has the benefit of the rrev value not changing unless the recipe is changed, whereas the Git registry_version qualifier changes every time any recipe is added or removed or modified, but with the drawback of it being much more difficult to resolve that rrev to its recipe file without maintaining an index.

Reply:
conan also supports [revision](https://docs.conan.io/2/reference/conanfile_txt.html#requires) btw.

Reply:
> However, instead of a Git commit ID it's a hash of the recipe files.

This would be the case for vcpkg as well.

Reply:
Any thoughts on calling the qualifier revision instead of registry_version?

Reply:
This is renamed to `repository_revision` now, to distinguish it from `port_revision`. Both have been explicitly described to clarify the difference.